### ISSUE TITLE
Mix.Config.Agent use is incompatible with Elixir master

### ISSUE THREAD

**Issue by ivan** [bitwalker/distillery#431]
### Steps to reproduce

Use Elixir master, distillery master, run a `mix release` and see something like

```
** (Mix.Releases.Config.LoadError) could not load release config rel/config.exs
    ** (UndefinedFunctionError) function Mix.Config.Agent.start_link/0 is undefined (module Mix.Config.Agent is not available)
    Mix.Config.Agent.start_link()
    (stdlib) erl_eval.erl:677: :erl_eval.do_apply/6
    (stdlib) erl_eval.erl:446: :erl_eval.expr/5
    (stdlib) erl_eval.erl:126: :erl_eval.exprs/5
    (elixir) lib/code.ex:232: Code.eval_string/3
    lib/mix/lib/releases/config/config.ex:343: Mix.Releases.Config.read_string!/1
    lib/mix/lib/releases/config/config.ex:365: Mix.Releases.Config.read!/1
    lib/mix/lib/releases/config/config.ex:60: Mix.Releases.Config.get/1
    lib/distillery/tasks/release.ex:76: Mix.Tasks.Release.run/1
    (mix) lib/mix/task.ex:315: Mix.Task.run_task/3
    (mix) lib/mix/cli.ex:79: Mix.CLI.run_task/2
    (elixir) lib/code.ex:758: Code.require_file/2
```

distillery uses `Mix.Config.Agent` in https://github.com/bitwalker/distillery/blob/aa52390addb60d293e414dc157614004d3928208/lib/mix/lib/releases/config/config.ex#L34 while Elixir removed it in https://github.com/elixir-lang/elixir/commit/4fad9921a61e1b405a9cf86623c4cff2784d22ba

**Comment by bitwalker** [comment]
Thanks for the heads up! This should be relatively easy to address, I'll make sure that's done before pushing 2.0

**Comment by josevalim** [comment]
Right, the root issue here is that distillery is relying on private Mix APIs, and those may break/change at any time. I personally don't think evaluating Mix configs in a release is the way to go, since they may perform many assumptions related to the development and not about the host. However, we do need to introduce a proper mechanism for loading Elixir configuration without depending on Mix.

**Comment by bitwalker** [comment]
Fixed in master

**Comment by doughsay** [comment]
Elixir 1.7 is out now, so distillery releases don't work without this fix.  Do we need to wait for a 2.0 release, or can we get the relevant fixes in a 1.5.x patch release?  Or would they not be backwards compatible?

**Comment by fenollp** [comment]
@bitwalker Could you cut publish a release `≥ 1.5.3` to Hex with this patch pretty please? :)

**Comment by GalliGit** [comment]
@bitwalker When 2.0 will be available on hex?

**Comment by bitwalker** [comment]
It already is, in RC form, final release is this week

**Comment by DianaOlympos** [comment]
Is there any way we could get a fix in the 1.5 branch for this. Or at least a hard "no" we can link people to @bitwalker ? 

I totally understand if it is a hard no, but upgrading a major elixir version at the same time as a major dependency is problematic. Right now people will discover that problem when upgrading to 1.7, then have to do both at the same time, or at least revert some change, deal with distillery upgrade, then move back to elixir bump.

Having a 1.5 version of this fix would help tremendously by providing a smaller more incremental way to fix.

**Comment by bitwalker** [comment]
@DianaOlympos I've pushed the fix to the 1.5.x branch, and am pushing a new 1.5.4 release. I would still highly recommend upgrading as soon as you can, but I understand not wanting to change a bunch of stuff all at once :)
